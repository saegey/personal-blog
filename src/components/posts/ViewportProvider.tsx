import React from "react";

const viewportContext = React.createContext({ width: 0, height: 0 });

const ViewportProvider = ({
	children,
}: {
	children: JSX.Element;
}): JSX.Element => {
	const isBrowser = typeof window !== "undefined";
	if (!isBrowser) return <></>;
	const [width, setWidth] = React.useState(window.innerWidth);
	const [height, setHeight] = React.useState(window.innerHeight);

	const handleWindowResize = () => {
		setWidth(window.innerWidth);
		setHeight(window.innerHeight);
	};

	React.useEffect(() => {
		window.addEventListener("resize", handleWindowResize);
		return () => window.removeEventListener("resize", handleWindowResize);
	}, []);

	return (
		<viewportContext.Provider value={{ width, height }}>
			{children}
		</viewportContext.Provider>
	);
};

const useViewport = () => {
	const { width, height } = React.useContext(viewportContext);
	return { width, height };
};

export { useViewport };

export default ViewportProvider;
