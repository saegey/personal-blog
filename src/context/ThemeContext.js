import React from "react"

const defaultState = {
  dark: false,
  toggleDark: () => {},
}

const ThemeContext = React.createContext(defaultState)

// Getting dark mode information from OS!
// You need macOS Mojave + Safari Technology Preview Release 68 to test this currently.
const supportsDarkMode = () =>
  window.matchMedia("(prefers-color-scheme: dark)").matches === true

class ThemeProvider extends React.Component {
  state = {
    unitOfMeasure: false,
  }

  toggleUnit = () => {
    // console.log("oh hey")
    let unitOfMeasure
    if (this.state.unitOfMeasure === "imperial") {
      unitOfMeasure = "metric"
    } else {
      unitOfMeasure = "imperial"
    }
    localStorage.setItem("unitOfMeasure", unitOfMeasure)
    this.setState({ unitOfMeasure })
  }

  componentDidMount() {
    // Getting dark mode value from localStorage!
    const lsDark = localStorage.getItem("unitOfMeasure")
    if (lsDark) {
      this.setState({ unitOfMeasure: lsDark })
    } else if (supportsDarkMode()) {
      this.setState({ unitOfMeasure: "imperial" })
    }
  }

  render() {
    const { children } = this.props
    const { unitOfMeasure } = this.state
    return (
      <ThemeContext.Provider
        value={{
          unitOfMeasure,
          toggleUnit: this.toggleUnit,
        }}
      >
        {children}
      </ThemeContext.Provider>
    )
  }
}

export default ThemeContext

export { ThemeProvider }
