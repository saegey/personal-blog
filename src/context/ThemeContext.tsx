import { createContext, Component } from 'react'

const defaultState = {
  unitOfMeasure: 'imperial',
  toggleUnit: () => {},
}

const ThemeContext = createContext(defaultState)

class ThemeProvider extends Component {
  state = {
    unitOfMeasure: 'imperial',
  }

  toggleUnit = () => {
    console.log('unit toggle')
    let unitOfMeasure
    if (this.state.unitOfMeasure === 'imperial') {
      unitOfMeasure = 'metric'
    } else {
      unitOfMeasure = 'imperial'
    }
    localStorage.setItem('unitOfMeasure', unitOfMeasure)
    this.setState({ unitOfMeasure })
  }

  componentDidMount() {
    const unit = localStorage.getItem('unitOfMeasure')
    if (unit) {
      this.setState({ unitOfMeasure: unit })
    } else {
      this.setState({ unitOfMeasure: 'imperial' })
    }
  }

  render() {
    const { children } = this.props
    const { unitOfMeasure } = this.state
    return (
      <ThemeContext.Provider
        value={{
          unitOfMeasure,
          toggleUnit: this.toggleUnit,
        }}
      >
        {children}
      </ThemeContext.Provider>
    )
  }
}

export default ThemeContext

export { ThemeProvider }
