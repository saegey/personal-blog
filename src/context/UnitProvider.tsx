import { createContext, Component, useContext } from 'react'

const defaultState = {
  unitOfMeasure: 'imperial',
  distanceUnit: 'miles',
  elevationUnit: 'feet',
  toggleUnit: () => {},
}

const unitContext = createContext(defaultState)

class UnitProvider extends Component {
  state = {
    unitOfMeasure: 'imperial',
    distanceUnit: 'miles',
    elevationUnit: 'feet',
  }

  toggleUnit = () => {
    let unitOfMeasure, elevationUnit, distanceUnit
    if (this.state.unitOfMeasure === 'imperial') {
      unitOfMeasure = 'metric'
      elevationUnit = 'meters'
      distanceUnit = 'km'
    } else {
      unitOfMeasure = 'imperial'
      elevationUnit = 'feet'
      distanceUnit = 'miles'
    }
    localStorage.setItem('unitOfMeasure', unitOfMeasure)
    this.setState({ unitOfMeasure, elevationUnit, distanceUnit })
  }

  componentDidMount() {
    const unit = localStorage.getItem('unitOfMeasure')
    if (unit) {
      this.setState({ unitOfMeasure: unit })
    } else {
      this.setState({ unitOfMeasure: 'imperial' })
    }
  }

  render() {
    const { children } = this.props
    const { unitOfMeasure, distanceUnit, elevationUnit } = this.state
    return (
      <unitContext.Provider
        value={{
          unitOfMeasure,
          distanceUnit,
          elevationUnit,
          toggleUnit: this.toggleUnit,
        }}
      >
        {children}
      </unitContext.Provider>
    )
  }
}

const useUnits = () => {
  const { unitOfMeasure, distanceUnit, elevationUnit, toggleUnit } =
    useContext(unitContext)
  return { unitOfMeasure, distanceUnit, elevationUnit, toggleUnit }
}

export default UnitProvider

export { useUnits }
