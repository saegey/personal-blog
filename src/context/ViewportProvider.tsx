import { useState, createContext, useEffect, useContext } from 'react'

const viewportContext = createContext({})

const ViewportProvider = ({ children }: { children: JSX.Element }) => {
  const isBrowser = typeof window !== 'undefined'
  if (!isBrowser) return
  const [width, setWidth] = useState(window.innerWidth)
  const [height, setHeight] = useState(window.innerHeight)

  const handleWindowResize = () => {
    setWidth(window.innerWidth)
    setHeight(window.innerHeight)
  }

  useEffect(() => {
    window.addEventListener('resize', handleWindowResize)
    return () => window.removeEventListener('resize', handleWindowResize)
  }, [])

  return (
    <viewportContext.Provider value={{ width, height }}>
      {children}
    </viewportContext.Provider>
  )
}

const useViewport = () => {
  const { width, height } = useContext(viewportContext)
  return { width, height }
}

export { useViewport }

export default ViewportProvider
