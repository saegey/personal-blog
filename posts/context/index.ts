import UnitProvider from './UnitProvider';
import ViewportProvider from './ViewportProvider';

export { UnitProvider, ViewportProvider }
