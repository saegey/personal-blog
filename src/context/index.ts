import UnitProvider, { useUnits } from './UnitProvider'
import ViewportProvider, { useViewport } from './ViewportProvider'

export { UnitProvider, ViewportProvider, useUnits, useViewport }
