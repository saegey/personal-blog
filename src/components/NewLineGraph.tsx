import {
  AreaChart,
  Area,
  XAxis,
  YAxis,
  Tooltip,
  CartesianGrid,
  ResponsiveContainer,
} from 'recharts'
import { Box, useThemeUI } from 'theme-ui'

import { useViewport } from '../context/ViewportProvider'
import GradeGradient from './GradeGradient'
import { useUnits } from '../context/UnitProvider'

interface NewLineGraphProps {
  xMax: number
  downSampledData: Array<{
    x: number
    y: number
    distance: number
    color: string
  }>
  setMarker: (arg: any) => { }
  elevationToAdd: number
  axisLeftTickValues: {
    imperial: Array<Array<number>>
    metric: Array<Array<number>>
  }
  axisXTickValues: {
    imperial: Array<Array<number>>
    metric: Array<Array<number>>
  }
  yMin: number
}

const NewLineGraph = ({
  xMax,
  downSampledData,
  setMarker,
  elevationToAdd = 0,
  axisLeftTickValues,
  axisXTickValues,
  yMin,
}: NewLineGraphProps) => {
  const themeContext = useThemeUI()
  const units = useUnits()

  const yTicks =
    units.unitOfMeasure === 'imperial'
      ? axisLeftTickValues.imperial[0]
      : axisLeftTickValues.metric[0]

  const xTicks =
    units.unitOfMeasure === 'imperial'
      ? axisXTickValues.imperial[0]
      : axisXTickValues.metric[0]

  const { width } = useViewport()
  const hideAxes = width > 640 ? false : true

  return (
    <Box
      sx={{
        width: '100%',
        height: ['100px', '200px', '250px'],
        borderColor: 'mutedAccent',
        borderStyle: 'solid',
        borderWidth: '1px',
        paddingY: [0, '20px', '20px'],
        paddingRight: [0, '20px', '20px'],
      }}
    >
      <ResponsiveContainer width={'100%'} height={"100%"}>
        <AreaChart
          data={downSampledData}
          onMouseMove={e => {
            if (!e || !e.activePayload) {
              setMarker(null)
              return
            }

            setMarker(e.activePayload[0].payload)
          }}
          margin={{ top: 0, right: 0, left: 0, bottom: 0 }}
        >
          {!hideAxes && (
            <CartesianGrid stroke={String(themeContext.theme.colors?.muted)} />
          )}

          <Tooltip content={<></>} />
          <defs>
            <linearGradient id="splitColor" x1="0" y1="0" x2="1" y2="0">
              <GradeGradient data={downSampledData} xMax={xMax} />
            </linearGradient>
          </defs>
          <XAxis
            dataKey="distance"
            type={'number'}
            ticks={xTicks}
            domain={[0, xMax]}
            hide={hideAxes}
          />
          <YAxis
            type="number"
            domain={[
              units.unitOfMeasure === 'imperial' ? yMin : yMin * 0.3048,
              `dataMax + ${
                units.unitOfMeasure === 'imperial'
                  ? elevationToAdd
                  : elevationToAdd * 0.3048
              }`,
            ]}
            ticks={yTicks}
            hide={hideAxes}
          />
          <Area
            type="basisOpen"
            dataKey="y"
            stroke="url(#splitColor)"
            strokeWidth={3}
            fill="gray"
            fillOpacity={0.1}
            dot={false}
          />
        </AreaChart>
      </ResponsiveContainer>
    </Box>
  )
}

export default NewLineGraph
