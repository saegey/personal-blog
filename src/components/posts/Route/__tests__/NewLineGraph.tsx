import { render } from '@testing-library/react'
import NewLineGraph from '../NewLineGraph'
import { useViewport } from '../../../../context/ViewportProvider'
import { mocked } from 'jest-mock'

jest.mock('recharts', () => {
  const OriginalModule = jest.requireActual('recharts')
  return {
    ...OriginalModule,
    ResponsiveContainer: ({ children }: { children: JSX.Element }) => (
      <OriginalModule.ResponsiveContainer width={800} height={800}>
        {children}
      </OriginalModule.ResponsiveContainer>
    ),
  }
})

jest.mock('../../../../context/UnitProvider.tsx', () => {
  return {
    useUnits: () => {
      return { unitOfMeasure: 'imperial' }
    },
  }
})

jest.mock('../../../../context/ViewportProvider.tsx', () => {
  return {
    useViewport: jest.fn(),
  }
})

const mockedUseViewport = mocked(useViewport, true)

const data = {
  xMax: 500,
  downSampledData: [
    { x: 10, y: 200, distance: 1, color: 'red' },
    { x: 20, y: 220, distance: 2, color: 'yellow' },
    { x: 30, y: 240, distance: 3, color: 'blue' },
  ],
  elevationToAdd: 100,
  setMarker: jest.fn(),
  axisLeftTickValues: {
    imperial: [[0, 10, 20, 50]],
    metric: [[0, 10, 20, 50]],
  },
  axisXTickValues: {
    imperial: [[0, 10, 20, 50]],
    metric: [[0, 10, 20, 50]],
  },
  yMin: 0,
}

describe('NewLineGraph', () => {
  it('renders correctly', () => {
    mockedUseViewport.mockReturnValueOnce({ width: 800, height: 600 })
    const screen = render(
      <NewLineGraph
        downSampledData={data.downSampledData}
        xMax={data.xMax}
        axisLeftTickValues={data.axisLeftTickValues}
        axisXTickValues={data.axisXTickValues}
        yMin={data.yMin}
        elevationToAdd={data.elevationToAdd}
        setMarker={data.setMarker}
      />
    )
    expect(screen.container).toMatchSnapshot()
  })

  it('doesnt render axes if width is less than 640', () => {
    mockedUseViewport.mockReturnValueOnce({ width: 600, height: 400 })
    const screen = render(
      <NewLineGraph
        downSampledData={data.downSampledData}
        xMax={data.xMax}
        axisLeftTickValues={data.axisLeftTickValues}
        axisXTickValues={data.axisXTickValues}
        yMin={data.yMin}
        elevationToAdd={data.elevationToAdd}
        setMarker={data.setMarker}
      />
    )
    expect(screen.container).toMatchSnapshot()
  })
})
