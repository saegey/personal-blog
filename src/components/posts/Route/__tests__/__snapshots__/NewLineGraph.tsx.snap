// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewLineGraph doesnt render axes if width is less than 640 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  height: 100px;
  border-color: mutedAccent;
  border-style: solid;
  border-width: 1px;
  padding-top: 0;
  padding-bottom: 0;
  padding-right: 0;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    height: 200px;
    padding-top: 20px;
    padding-bottom: 20px;
    padding-right: 20px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-0 {
    height: 250px;
    padding-top: 20px;
    padding-bottom: 20px;
    padding-right: 20px;
  }
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="recharts-responsive-container"
      style="width: 800px; height: 800px; min-width: 0;"
    >
      <div
        class="recharts-wrapper"
        role="region"
        style="position: relative; cursor: default; width: 800px; height: 800px;"
      >
        <svg
          class="recharts-surface"
          height="800"
          viewBox="0 0 800 800"
          width="800"
        >
          <title />
          <desc />
          <defs>
            <clippath
              id="recharts4-clip"
            >
              <rect
                height="800"
                width="800"
                x="0"
                y="0"
              />
            </clippath>
          </defs>
          <defs>
            <lineargradient
              id="splitColor"
              x1="0"
              x2="1"
              y1="0"
              y2="0"
            >
              <stop
                offset="0.002"
                stop-color="red"
                stop-opacity="1"
              />
              <stop
                offset="0.004"
                stop-color="yellow"
                stop-opacity="1"
              />
              <stop
                offset="0.006"
                stop-color="blue"
                stop-opacity="1"
              />
            </lineargradient>
          </defs>
          <g
            class="recharts-layer recharts-area"
          >
            <g
              class="recharts-layer"
            >
              <defs>
                <clippath
                  id="animationClipPath-recharts-area-6"
                >
                  <rect
                    height="803"
                    width="0"
                    x="1.6"
                    y="0"
                  />
                </clippath>
              </defs>
              <g
                class="recharts-layer"
                clip-path="url(#animationClipPath-recharts-area-6)"
              >
                <g
                  class="recharts-layer"
                >
                  <path
                    class="recharts-curve recharts-area-area"
                    d="M3.1999999999999997,282.35294117647055L3.2000000000000006,800Z"
                    fill="gray"
                    fill-opacity="0.1"
                    height="800"
                    stroke="none"
                    stroke-width="3"
                    width="800"
                  />
                  <path
                    class="recharts-curve recharts-area-curve"
                    d="M3.1999999999999997,282.35294117647055Z"
                    fill="none"
                    fill-opacity="0.1"
                    height="800"
                    stroke="url(#splitColor)"
                    stroke-width="3"
                    width="800"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
        <div
          class="recharts-tooltip-wrapper"
          role="dialog"
          style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
          tabindex="-1"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NewLineGraph renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  height: 100px;
  border-color: mutedAccent;
  border-style: solid;
  border-width: 1px;
  padding-top: 0;
  padding-bottom: 0;
  padding-right: 0;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    height: 200px;
    padding-top: 20px;
    padding-bottom: 20px;
    padding-right: 20px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-0 {
    height: 250px;
    padding-top: 20px;
    padding-bottom: 20px;
    padding-right: 20px;
  }
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="recharts-responsive-container"
      style="width: 800px; height: 800px; min-width: 0;"
    >
      <div
        class="recharts-wrapper"
        role="region"
        style="position: relative; cursor: default; width: 800px; height: 800px;"
      >
        <svg
          class="recharts-surface"
          height="800"
          viewBox="0 0 800 800"
          width="800"
        >
          <title />
          <desc />
          <defs>
            <clippath
              id="recharts1-clip"
            >
              <rect
                height="770"
                width="740"
                x="60"
                y="0"
              />
            </clippath>
          </defs>
          <g
            class="recharts-cartesian-grid"
          >
            <g
              class="recharts-cartesian-grid-horizontal"
            >
              <line
                fill="none"
                height="770"
                offset="[object Object]"
                stroke="undefined"
                width="740"
                x="60"
                x1="60"
                x2="800"
                y="0"
                y1="770"
                y2="770"
              />
              <line
                fill="none"
                height="770"
                offset="[object Object]"
                stroke="undefined"
                width="740"
                x="60"
                x1="60"
                x2="800"
                y="0"
                y1="747.3529411764706"
                y2="747.3529411764706"
              />
              <line
                fill="none"
                height="770"
                offset="[object Object]"
                stroke="undefined"
                width="740"
                x="60"
                x1="60"
                x2="800"
                y="0"
                y1="724.7058823529412"
                y2="724.7058823529412"
              />
              <line
                fill="none"
                height="770"
                offset="[object Object]"
                stroke="undefined"
                width="740"
                x="60"
                x1="60"
                x2="800"
                y="0"
                y1="656.7647058823529"
                y2="656.7647058823529"
              />
              <line
                fill="none"
                height="770"
                offset="[object Object]"
                stroke="undefined"
                width="740"
                x="60"
                x1="60"
                x2="800"
                y="0"
                y1="0"
                y2="0"
              />
            </g>
            <g
              class="recharts-cartesian-grid-vertical"
            >
              <line
                fill="none"
                height="770"
                offset="[object Object]"
                stroke="undefined"
                width="740"
                x="60"
                x1="60"
                x2="60"
                y="0"
                y1="0"
                y2="770"
              />
              <line
                fill="none"
                height="770"
                offset="[object Object]"
                stroke="undefined"
                width="740"
                x="60"
                x1="74.8"
                x2="74.8"
                y="0"
                y1="0"
                y2="770"
              />
              <line
                fill="none"
                height="770"
                offset="[object Object]"
                stroke="undefined"
                width="740"
                x="60"
                x1="89.6"
                x2="89.6"
                y="0"
                y1="0"
                y2="770"
              />
              <line
                fill="none"
                height="770"
                offset="[object Object]"
                stroke="undefined"
                width="740"
                x="60"
                x1="134"
                x2="134"
                y="0"
                y1="0"
                y2="770"
              />
              <line
                fill="none"
                height="770"
                offset="[object Object]"
                stroke="undefined"
                width="740"
                x="60"
                x1="800"
                x2="800"
                y="0"
                y1="0"
                y2="770"
              />
            </g>
          </g>
          <defs>
            <lineargradient
              id="splitColor"
              x1="0"
              x2="1"
              y1="0"
              y2="0"
            >
              <stop
                offset="0.002"
                stop-color="red"
                stop-opacity="1"
              />
              <stop
                offset="0.004"
                stop-color="yellow"
                stop-opacity="1"
              />
              <stop
                offset="0.006"
                stop-color="blue"
                stop-opacity="1"
              />
            </lineargradient>
          </defs>
          <g
            class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
          >
            <line
              class="recharts-cartesian-axis-line"
              fill="none"
              height="30"
              orientation="bottom"
              stroke="#666"
              width="740"
              x="60"
              x1="60"
              x2="800"
              y="770"
              y1="770"
              y2="770"
            />
            <g
              class="recharts-cartesian-axis-ticks"
            >
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="30"
                  orientation="bottom"
                  stroke="#666"
                  width="740"
                  x="60"
                  x1="60"
                  x2="60"
                  y="770"
                  y1="776"
                  y2="770"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  text-anchor="middle"
                  width="740"
                  x="60"
                  y="778"
                >
                  <tspan
                    dy="0.71em"
                    x="60"
                  >
                    0
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="30"
                  orientation="bottom"
                  stroke="#666"
                  width="740"
                  x="60"
                  x1="74.8"
                  x2="74.8"
                  y="770"
                  y1="776"
                  y2="770"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  text-anchor="middle"
                  width="740"
                  x="74.8"
                  y="778"
                >
                  <tspan
                    dy="0.71em"
                    x="74.8"
                  >
                    10
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="30"
                  orientation="bottom"
                  stroke="#666"
                  width="740"
                  x="60"
                  x1="89.6"
                  x2="89.6"
                  y="770"
                  y1="776"
                  y2="770"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  text-anchor="middle"
                  width="740"
                  x="89.6"
                  y="778"
                >
                  <tspan
                    dy="0.71em"
                    x="89.6"
                  >
                    20
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="30"
                  orientation="bottom"
                  stroke="#666"
                  width="740"
                  x="60"
                  x1="134"
                  x2="134"
                  y="770"
                  y1="776"
                  y2="770"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  text-anchor="middle"
                  width="740"
                  x="134"
                  y="778"
                >
                  <tspan
                    dy="0.71em"
                    x="134"
                  >
                    50
                  </tspan>
                </text>
              </g>
            </g>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
          >
            <line
              class="recharts-cartesian-axis-line"
              fill="none"
              height="770"
              orientation="left"
              stroke="#666"
              width="60"
              x="0"
              x1="60"
              x2="60"
              y="0"
              y1="0"
              y2="770"
            />
            <g
              class="recharts-cartesian-axis-ticks"
            >
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="770"
                  orientation="left"
                  stroke="#666"
                  width="60"
                  x="0"
                  x1="54"
                  x2="60"
                  y="0"
                  y1="770"
                  y2="770"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="770"
                  orientation="left"
                  stroke="none"
                  text-anchor="end"
                  width="60"
                  x="52"
                  y="770"
                >
                  <tspan
                    dy="0.355em"
                    x="52"
                  >
                    0
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="770"
                  orientation="left"
                  stroke="#666"
                  width="60"
                  x="0"
                  x1="54"
                  x2="60"
                  y="0"
                  y1="747.3529411764706"
                  y2="747.3529411764706"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="770"
                  orientation="left"
                  stroke="none"
                  text-anchor="end"
                  width="60"
                  x="52"
                  y="747.3529411764706"
                >
                  <tspan
                    dy="0.355em"
                    x="52"
                  >
                    10
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="770"
                  orientation="left"
                  stroke="#666"
                  width="60"
                  x="0"
                  x1="54"
                  x2="60"
                  y="0"
                  y1="724.7058823529412"
                  y2="724.7058823529412"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="770"
                  orientation="left"
                  stroke="none"
                  text-anchor="end"
                  width="60"
                  x="52"
                  y="724.7058823529412"
                >
                  <tspan
                    dy="0.355em"
                    x="52"
                  >
                    20
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="770"
                  orientation="left"
                  stroke="#666"
                  width="60"
                  x="0"
                  x1="54"
                  x2="60"
                  y="0"
                  y1="656.7647058823529"
                  y2="656.7647058823529"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="770"
                  orientation="left"
                  stroke="none"
                  text-anchor="end"
                  width="60"
                  x="52"
                  y="656.7647058823529"
                >
                  <tspan
                    dy="0.355em"
                    x="52"
                  >
                    50
                  </tspan>
                </text>
              </g>
            </g>
          </g>
          <g
            class="recharts-layer recharts-area"
          >
            <g
              class="recharts-layer"
            >
              <defs>
                <clippath
                  id="animationClipPath-recharts-area-3"
                >
                  <rect
                    height="773"
                    width="0"
                    x="61.480000000000004"
                    y="0"
                  />
                </clippath>
              </defs>
              <g
                class="recharts-layer"
                clip-path="url(#animationClipPath-recharts-area-3)"
              >
                <g
                  class="recharts-layer"
                >
                  <path
                    class="recharts-curve recharts-area-area"
                    d="M62.96,271.7647058823529L62.96,770Z"
                    fill="gray"
                    fill-opacity="0.1"
                    height="770"
                    stroke="none"
                    stroke-width="3"
                    width="740"
                  />
                  <path
                    class="recharts-curve recharts-area-curve"
                    d="M62.96,271.7647058823529Z"
                    fill="none"
                    fill-opacity="0.1"
                    height="770"
                    stroke="url(#splitColor)"
                    stroke-width="3"
                    width="740"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
        <div
          class="recharts-tooltip-wrapper"
          role="dialog"
          style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
          tabindex="-1"
        />
      </div>
    </div>
  </div>
</div>
`;
